### ISSUE TITLE
WAVE Toolbar for Firefox has been discontinued

### ISSUE THREAD

**Issue by datakurre** [datakurre/robotframework-selenium2accessibility#1]
https://wave.webaim.org/toolbar/

Possibly, https://addons.mozilla.org/en-US/firefox/addon/wcag-contrast-checker/ is also broken with the current selenium2accessibility master, but fixing it won't happen if WAVE support cannot be fixed.

Probably, I need to figure out a way to make everything work with Chrome, which has supported extensions.

**Comment by emanuelsimon-qa** [comment]
Any updates on this matter?

**Comment by idgserpro** [comment]
@datakurre It appears that the extension has not been discontinued. See:

https://addons.mozilla.org/en-US/firefox/addon/wave-accessibility-tool/

**Comment by datakurre** [comment]
@idgserpro It has been completely rewritten. The version I used in this was for Firefox before ”Quantum” (removal of XUL) and replacing old extension framework with webextensions (spec shared by browser vendors).

I did look into this a bit last fall, but did not find any supported way to call the current extension through the sandbox. If there were a way for an another extension to call methods of an another extension, this could be done again. Even a way to programmatically opening Wave sidebar from an another extension could work. Haven’t figured that out yet, though.

**Comment by datakurre** [comment]
It might be that I only tested if it is possible to somehow open the sidebar directly from content window. Not if it would be possible to open it from a custom extension that could somehow be called from content window. So maybe there is still change to bring this alive again even in this more restricted web extension era :thinking:

**Comment by datakurre** [comment]
I experimented on this.

The current WAVE-extension cannot be automated without customizing the extension itself. That's because of how web extension security has been designed (to be secure).

That said, with small customization, WAVE-extension can be automated with Robot Framework and Selenium. And internally it has good enough API and excellent results format.

But because of the need to customize WAVE-extension, I may not be able to fix this package :(
